### PR TITLE
V2 DXF Deserializer

### PR DESCRIPTION
**Pull request by z3dev**
This pull request updates the SVG deserializer to use CSG V2 API, as well as translate to CSG V2 API. The goals of this pull request are:
- [X] Update to use CSG V2 API
- [X] Update code up to JSCAD coding standards
- [X] Add test suites to complete coverage, 88.69% (93.11% instantiate, 71.22% translate)

New Features:
- added support for colors on paths and 2D geometry

Fixes issue jscad/io#83 
Fixed issues with translating bulges in 2D paths
Fixed issues with translating successive polylines
Fixed issues with closing polylines

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

File: packages/io/dxf-deserializer/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+node_modules
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/.gitignore]: .gitignore is not needed :)

File: packages/io/dxf-deserializer/README.md [hunk 2]
```diff
@@ -1,36 +1,37 @@
 ## @jscad/dxf-deserializer
 
-> DXF deserializer for the JSCAD project
+> Deserializer of DXF data to CSG geometries
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/README.md]: just geometries :)

File: packages/io/dxf-deserializer/README.md [hunk 3]
```diff
@@ -1,36 +1,37 @@
...
+When converting unknown DXF data, converting to JSCAD scripts will provide more information on the converted DXF entities.
 
 ### DXF Implementation Notes
 
 The [DXF(tm) file structure](http://docs.autodesk.com/ACD/2014/ENU/files/GUID-73E9E797-3BAA-4795-BBD8-4CE7A03E93CF.htm) changes continously as AutoDesk releases new features and fixes. So, you can imagine that after 20 years, the contents of DXF files are pretty messed up. Expect the worst.
 
 **NOTE: At this time, only ASCII DXF files are supported. BINARY DXF files are not supported.**
 
-This deserializer converts only what is required by JSCAD libraries and applications. Full document conversion is NOT supported, so don't even try. However, conversion of the following DXF entities are possible:
+This deserializer converts only what is supported by JSCAD libraries. Full document conversion is NOT supported, so don't even try. However, conversion of the following DXF entities are possible:
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/README.md]: the 'so don't even try part' is kind off odd, tone wise ?

File: packages/io/dxf-deserializer/README.md [hunk 4]
```diff
@@ -1,36 +1,37 @@
...
-| ARC             | to Path2D  | |
-| CIRCLE          | to CAG     | Start and stop angles are ignored |
-| ELLIPSE         | to CAG     | Start and stop angles are ignored |
-| LINE            | to Line2D or Line3D | This will change in the future |
-| LWPOLYLINE      | to Path2D or CAG  | Conversion to CAG if LWPOLYLINE is closed |
-| MESH            | to CSG     | |
-| POLYLINE (line) | to Path2D or CAG | Conversion to CAG if POLYLINE is closed |
-| POLYLINE (mesh) | to CSG     | |
-| POLYLINE (face) | to CSG     | |
+| DXF Entity      | CSG Geometry | Notes |
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/README.md]: Don't we have more stuff working than before in V2 ? :)

File: packages/io/dxf-deserializer/README.md [hunk 5]
```diff
@@ -51,18 +51,18 @@ npm install @jscad/dxf-deserializer
 ## Usage
 
 ```javascript
-const deSerializer = require('@jscad/dxf-deserializer')
+const dxfDeserializer = require('@jscad/dxf-deserializer')
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/README.md]: nice catch !

File: packages/io/dxf-deserializer/instantiate.js [hunk 6]
```diff
@@ -207,79 +221,29 @@ function instantiateEllipse (obj, layers, options) {
...
-        if (v1edges.indexOf(v3) < 0) continue
-        faces.push([v1, v2, v3])
-      }
-    }
-  }
-  return faces
-}
-
-function instantiateFaces (fvals) {
+const instantiateFaces = (fvals) => {
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/instantiate.js]: no need for the rest anymore ?

File: packages/io/dxf-deserializer/instantiate.js [hunk 7]
```diff
@@ -364,107 +324,70 @@ function instantiateMesh (obj, layers, options) {
   } else {
     // invalid vlen
   }
-  return CSG.fromPolygons(polygons)
-}
-
-//
-// find the layer referenced by the given object
-//
-function findLayer (obj, layers) {
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/instantiate.js]: not needed either ?

File: packages/io/dxf-deserializer/package.json [hunk 8]
```diff
@@ -29,9 +31,17 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "@jscad/csg": "^0.7.0"
+    "@jscad/csg": "0.7.0"
```
Comment (kaosat-dev) [inline packages/io/dxf-deserializer/package.json]: upgrade needed :)

File: packages/io/dxf-deserializer/README.md (hunk 4 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/README.md]: Nope. Basically the same. BUT there are some bug fixes. See the commit messages.

File: packages/io/dxf-deserializer/.gitignore (hunk 1 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/.gitignore]: Done

File: packages/io/dxf-deserializer/instantiate.js (hunk 7 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/instantiate.js]: Moved to helpers.js, which eliminates duplicate code.

File: packages/io/dxf-deserializer/package.json (hunk 8 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/package.json]: Done

File: packages/io/dxf-deserializer/README.md (hunk 2 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/README.md]: Done

File: packages/io/dxf-deserializer/README.md (hunk 3 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/README.md]: Done

File: packages/io/dxf-deserializer/instantiate.js (hunk 6 above)
Comment (z3dev) [reply packages/io/dxf-deserializer/instantiate.js]: Both createEdges() and createFaces() are unused. So.. bye bye!

**Comment by z3dev** [comment]
@kaosat-dev please review again, and merge if no futher changes are required. :)

File: packages/io/dxf-deserializer/instantiate.js (hunk 6 above)
Comment (kaosat-dev) [reply packages/io/dxf-deserializer/instantiate.js]: :) yeaah !!

File: packages/io/dxf-deserializer/README.md (hunk 4 above)
Comment (kaosat-dev) [reply packages/io/dxf-deserializer/README.md]: dang, ok thanks !

File: packages/io/dxf-deserializer/instantiate.js (hunk 7 above)
Comment (kaosat-dev) [reply packages/io/dxf-deserializer/instantiate.js]: ok good !

**Pull request merged**